Write new Task: Configuring a Pod to use a PersistentVolume.

steveperry-53 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2302)
<!-- Reviewable:end -->

pwittrock commented on docs/tasks/configure-pod-container/configure-persistent-volume-storage.md:
```diff
@@ -0,0 +1,196 @@
+## Creating an index.html file on your Node
+
+Open a shell to the Node in your cluster. How you open a shell depends on how
+you set up your cluster. For example, if you are using Minikube, you can open a
```
does minikube support PersistentVolumes?

pwittrock replied:
nevermind, answered

pwittrock commented on docs/tasks/configure-pod-container/configure-persistent-volume-storage.md:
```diff
@@ -0,0 +1,196 @@
+The next step is to create a PersistentVolumeClaim. Pods use PersistentVolumeClaims
+to request physical storage. In this exercise, you create a PersistentVolumeClaim
+that requests a volume of at least three gibibytes that can provide read-write
+access for at least one Node.
```
>  read-write access for at least one Node.

What does it meant to provide read-write access for the Node?  Should this be Pod, or is it something else?

steveperry-53 replied:
I need to understand this better. But here's a starting point: https://kubernetes.io/docs/user-guide/persistent-volumes/#access-modes.

steveperry-53 replied:
OK as is, I think.

pwittrock replied:
@saad-ali Would you comment on this?  Sounds like it is correct as written, but that seems like it may be confusing to users since we tell them to think in terms of Pods, not machines for most things.  Maybe we should add a discussion topic at the end for the nuances of the multi-mount behavior?

pwittrock commented on docs/tasks/configure-pod-container/configure-persistent-volume-storage.md:
```diff
@@ -0,0 +1,196 @@
+
+    echo 'Hello from Kubernetes storage' > /tmp/data/index.html
+
+## Creating a PersistentVolume
```
I forgot how complicated this all is.  Does it make sense to quickly provide an overview of the moving pieces and their relationship with one another?  My first read through this, it wasn't clear to me why all these things existed.

Roughly:
- PersistentVolume: Unallocated persistent storage provisioned by cluster admins that can be allocated.
- PersistentVolumeClaim: Allocation of persistent storage defined by a PersistentVolume.  Can be referenced within a Pod to mount storage as a Volume.
- Pod.Volume: Binds pre-allocated persistent storage for use within the Pod.

steveperry-53 replied:
I'll take a stab at putting some text at the top of the document.

steveperry-53 replied:
I added some text to the introduction.

pwittrock commented on docs/tasks/configure-pod-container/configure-persistent-volume-storage.md:
```diff
@@ -0,0 +1,196 @@
+## Creating a PersistentVolume
+
+In this exercise, you create a *hostPath* PersistentVolume. Kubernetes supports
+hostPath for development and testing on a single-node cluster. A hostPath
```
> uses a file or directory on the Node that is running the Pod.

WDYT of

uses a file or directory on the Node to emulate network attached storage.

steveperry-53 replied:
Sounds good.

steveperry-53 replied:
Done.

pwittrock commented on docs/tasks/configure-pod-container/configure-persistent-volume-storage.md:
```diff
@@ -0,0 +1,196 @@
+
+    kubectl get pv task-pv-volume
+
+The output shows that the PersistentVolume has a `STATUS` of `Available`. This
```
> claimed by a Pod.

The Pod doesn't claim it right?  IIRC, the PVC does.

steveperry-53 replied:
Not sure. I'll have to think about it.

steveperry-53 replied:
Updated the text.

pwittrock replied:
Sounds good.  Rethinking this, I suppose a Pod does claim the Volume using a PVC, but the STATUS is independent of whether a Pod is using the claim or not.

pwittrock commented on docs/tasks/configure-pod-container/configure-persistent-volume-storage.md:
```diff
@@ -0,0 +1,196 @@
+Storage configured with a group ID (GID) allows writing only by Pods using the same
+GID. Mismatched or missing GIDs cause permission denied errors. To reduce the
+need for coordination with users, an administrator can annotate a PersistentVolume
+with a GID. Then the GID is automatically added to any Pod that uses the
```
How does the Pod define that it uses the PersistentVolume?

steveperry-53 replied:
The Pod specifies a PVC. Behind the scenes, the PVC has been bound to a PV that fits the requirements. Is that not clear in the text?

pwittrock replied:
What you said is clear in the text, but something about it was surprising to me when I read it.  I think because in the rest of the document we say the Pod uses the PersistentVolumeClaim, but don't say it uses the Volume.  This is fine.  An alternative for your consideration:

>
added to any Pod that has claimed the PersistentVolume

pwittrock commented:
Review pass finished.

steveperry-53 commented:
@pwittrock, I addressed your comments in [this commit](https://github.com/kubernetes/kubernetes.github.io/pull/2302/commits/ee3ca9452712877ec9fac0c7f7bbdccf7ce6cc0a).

pwittrock commented:
LGTM pending Saad's input

steveperry-53 commented:
@saad-ali, Would you be able to take a look at this today?

saad-ali commented:
Thanks @steveperry-53. This looks good. What it make sense to mention dynamic provisioning (http://blog.kubernetes.io/2016/10/dynamic-provisioning-and-storage-in-kubernetes.html) somewhere in this doc?

steveperry-53 commented:
@saad-ali, I added a note about dynamic provisioning.
